src/session/user.rs

```diff
@@ -146,12 +147,88 @@ mod imp {
 
             let avatar = obj.avatar();
             let user_expression = gtk::ConstantExpression::new(obj);
-            let full_name_expression = super::User::full_name_expression(&user_expression);
-            full_name_expression.bind(avatar, "display-name", gtk::NONE_WIDGET);
+            let display_name_expression = super::User::display_name_expression(&user_expression);
+            display_name_expression.bind(avatar, "display-name", gtk::NONE_WIDGET);
         }
     }
 }
 
+fn last_seen_time_ago(unix: i32) -> String {
+    let datetime_now = glib::DateTime::new_now_local().unwrap();
+    let unix: i64 = unix.into();
+    let last_seen = glib::DateTime::from_unix_utc(unix)
+        .unwrap()
+        .to_local()
+        .unwrap();
+
+    let seconds = datetime_now.difference(&last_seen) / 1_000_000;
+    let minutes = seconds / 60;
+    let hours = minutes / 60;
+    if minutes < 120 {
+        if minutes == 0 {
+            gettext("last seen just now")
+        } else if minutes == 1 {
+            gettext("last seen a minute ago")
+        } else if minutes < 60 {
+            gettext!("last seen {} minutes ago", minutes)
+
+        // greater or equal to 60 minutes
+        } else {
+            gettext("last seen an hour ago")
+        }
+    } else if (
+        datetime_now.year(),
+        datetime_now.month(),
+        datetime_now.day_of_month(),
+    ) == (
+        last_seen.year(),
+        last_seen.month(),
+        last_seen.day_of_month(),
+    ) {
+        gettext!("last seen {} hours ago", hours)
+    } else if (
+        datetime_now.year(),
+        datetime_now.month(),
+        datetime_now.day_of_month(),
+    ) == (
+        last_seen.year(),
+        last_seen.month(),
+        last_seen.add_days(1).unwrap().day_of_month(),
```

Comment:
It seems you are making a logic error here.

Small example: 
Suppose `last_seen` was on __2021/12/31__, and `datetime_now` was on __2022/01/01__. Then you would compare `(2022, 1, 1) == (2021, 12, (31 + 1) % 31 = 1)`. But actually you want to compare `(2022, 1, 1) == (2022, 1, 1)`. To do this, you must add the days to the __whole__ date, and not just within the tuple.

Reply:
You're right it should display `yesterday` instead of full date in that case

Reply:
My next idea is to combine both time and date comparasing
```

Reply:
I'll close this and will continue discussion in another thread